server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/ConfigurationService.java

```diff
@@ -192,6 +192,8 @@
      */
     ConfigurationSchemaDto saveConfSchema(ConfigurationSchemaDto configurationSchema);
 
+    String validateConfiguration(String appId, int schemaVersion, String configurationBody);
```

Comment:
This method may return modified configuration (see UuidValidator used in method body). So it can not be void. We can rename it to normalizeAccordingToOverrideConfigurationSchema.